paseos/visualization/space_animation.py

```diff
@@ -0,0 +1,280 @@
+from paseos.paseos import PASEOS
+from paseos.visualization.animation import Animation
+
+
+class SpaceAnimation(Animation):
+    """This class visualizes the central body, local actor and known actors over time."""
+
+    def __init__(self, sim: PASEOS) -> None:
+
+        super().__init__(sim)
+        logger.trace("Initializing animation")
+        # Create list of objects to be plotted
+        current_actors = self._make_actor_list(sim)
+        self._norm_coeff = self._local_actor._central_body.radius
+
+        local_time_d = self._local_actor.local_time.mjd2000
+        for known_actor in current_actors:
+            pos, _ = known_actor.get_position_velocity(pk.epoch(local_time_d))
+            pos_norm = [x / self._norm_coeff for x in pos]
+            self.objects.append(DotMap(actor=known_actor, positions=np.array(pos_norm)))
+
+        # create figure
+        self.fig = plt.figure(figsize=plt.figaspect(0.5) * 1.5)
+        self.ax = plt.subplot(projection="3d")
+        self.ax.set_title(self._sec_to_ddhhmmss(local_time_d / pk.SEC2DAY))
+        self.ax.get_xaxis().set_ticks([])
+        self.ax.get_yaxis().set_ticks([])
+        self.ax.get_zaxis().set_ticks([])
+        self.n_trajectory = 50  # how many samples from histories to visualize
+        self._textbox_offset = 0.25  # how many samples from histories to visualize
+
+        # plot the objects
+        self._plot_central_body()
+        self._plot_actors()
+
+        plt.show(block=False)
+
+    def _plot_central_body(self) -> None:
+        """Plot the central object as a sphere of radius 1
+        Args:
+            None
+        Returns:
+            None
+        """
+        central_body = self._local_actor._central_body
+        central_body.radius
+
+        u, v = np.mgrid[0:2 * np.pi:30j, 0:np.pi:20j]
+        x = np.cos(u) * np.sin(v)
+        y = np.sin(u) * np.sin(v)
+        z = np.cos(v)
+        self.ax.plot_surface(x, y, z, color="grey")
+
+    def _populate_textbox(self, actor: BaseActor) -> str:
+        """Extracts information from an actor and builds a string
+        Args:
+            actor (BaseActor): an actor
+        Returns:
+            str: text to populate the textbox.
+        """
+        if isinstance(actor, SpacecraftActor):
+            # TODO: enable both text and icons in textbox
+            battery_level = actor.battery_level_ratio * 100
+            info_str = f"{actor.name} \nbat: {battery_level:.0f}%"
+            for name in actor.communication_devices.keys():
+                info = actor.communication_devices[name]
+                info_str += f"\n{name}: {info.bandwidth_in_kbps} kbps"
+
+        elif isinstance(actor, GroundstationActor):
+            # TODO: implement textbox for groundstation
+            pass
+
+        return info_str
+
+    def _plot_actors(self) -> None:
+        """Plots all the actors
+        Args:
+            None
+        Returns:
+            None
```

Comment:
Just remove, only if not None

I would recommend to use https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring it does it automatically for you

Reply:
Nice, didn't know about it. In fact, didn't know Sphinx is so sensitive. :)